File: dmn-core/src/test/resources/tck/cl3/expected/dmn/0004-lending/type/TApplicantData.java
package type;

import java.util.*;

@javax.annotation.Generated(value = {"itemDefinitionInterface.ftl", "tApplicantData"})
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
public interface TApplicantData extends com.gs.dmn.runtime.DMNType {
    static TApplicantData toTApplicantData(Object other) {
        if (other == null) {
            return null;
        } else if (TApplicantData.class.isAssignableFrom(other.getClass())) {
            return (TApplicantData)other;
        } else if (other instanceof com.gs.dmn.runtime.Context) {
            TApplicantDataImpl result_ = new TApplicantDataImpl();
            result_.setMonthly((type.Monthly)((com.gs.dmn.runtime.Context)other).get("Monthly"));
            result_.setAge((java.math.BigDecimal)((com.gs.dmn.runtime.Context)other).get("Age"));
            result_.setExistingCustomer((Boolean)((com.gs.dmn.runtime.Context)other).get("ExistingCustomer"));
            result_.setMaritalStatus((String)((com.gs.dmn.runtime.Context)other).get("MaritalStatus"));
            result_.setEmploymentStatus((String)((com.gs.dmn.runtime.Context)other).get("EmploymentStatus"));
            return result_;
        } else if (other instanceof com.gs.dmn.runtime.DMNType) {
            return toTApplicantData(((com.gs.dmn.runtime.DMNType)other).toContext());
        } else {
            throw new com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.getClass().getSimpleName(), TApplicantData.class.getSimpleName()));
        }
    }

    @com.fasterxml.jackson.annotation.JsonGetter("Monthly")
    type.Monthly getMonthly();

    @com.fasterxml.jackson.annotation.JsonGetter("Age")
    java.math.BigDecimal getAge();

    @com.fasterxml.jackson.annotation.JsonGetter("ExistingCustomer")
    Boolean getExistingCustomer();

    @com.fasterxml.jackson.annotation.JsonGetter("MaritalStatus")
    String getMaritalStatus();

    @com.fasterxml.jackson.annotation.JsonGetter("EmploymentStatus")
    String getEmploymentStatus();

    default com.gs.dmn.runtime.Context toContext() {
        com.gs.dmn.runtime.Context context = new com.gs.dmn.runtime.Context();
        context.put("monthly", getMonthly());
        context.put("age", getAge());
        context.put("existingCustomer", getExistingCustomer());
        context.put("maritalStatus", getMaritalStatus());
        context.put("employmentStatus", getEmploymentStatus());
        return context;
    }

    default boolean equalTo(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TApplicantData other = (TApplicantData) o;
        if (this.getAge() != null ? !this.getAge().equals(other.getAge()) : other.getAge() != null) return false;
        if (this.getEmploymentStatus() != null ? !this.getEmploymentStatus().equals(other.getEmploymentStatus()) : other.getEmploymentStatus() != null) return false;
        if (this.getExistingCustomer() != null ? !this.getExistingCustomer().equals(other.getExistingCustomer()) : other.getExistingCustomer() != null) return false;
        if (this.getMaritalStatus() != null ? !this.getMaritalStatus().equals(other.getMaritalStatus()) : other.getMaritalStatus() != null) return false;
        if (this.getMonthly() != null ? !this.getMonthly().equals(other.getMonthly()) : other.getMonthly() != null) return false;

        return true;
    }

    default int hash() {
        int result = 0;
        result = 31 * result + (this.getAge() != null ? this.getAge().hashCode() : 0);
        result = 31 * result + (this.getEmploymentStatus() != null ? this.getEmploymentStatus().hashCode() : 0);
        result = 31 * result + (this.getExistingCustomer() != null ? this.getExistingCustomer().hashCode() : 0);
        result = 31 * result + (this.getMaritalStatus() != null ? this.getMaritalStatus().hashCode() : 0);
        result = 31 * result + (this.getMonthly() != null ? this.getMonthly().hashCode() : 0);
        return result;
    }

    default String asString() {
        StringBuilder result_ = new StringBuilder("{");
        result_.append("Age=" + getAge());
        result_.append(", EmploymentStatus=" + getEmploymentStatus());
        result_.append(", ExistingCustomer=" + getExistingCustomer());
        result_.append(", MaritalStatus=" + getMaritalStatus());
        result_.append(", Monthly=" + getMonthly());
        result_.append("}");
        return result_.toString();
    }
}

File: dmn-core/src/test/resources/tck/cl3/expected/kotlin/dmn/0004-lending/type/TApplicantData.kt
package type

import java.util.*

@javax.annotation.Generated(value = ["itemDefinitionInterface.ftl", "tApplicantData"])
@com.fasterxml.jackson.annotation.JsonPropertyOrder(alphabetic = true)
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(`as` = type.TApplicantDataImpl::class)
interface TApplicantData : com.gs.dmn.runtime.DMNType {
    @get:com.fasterxml.jackson.annotation.JsonGetter("Monthly")
    val monthly: type.Monthly?

    @get:com.fasterxml.jackson.annotation.JsonGetter("Age")
    val age: java.math.BigDecimal?

    @get:com.fasterxml.jackson.annotation.JsonGetter("ExistingCustomer")
    val existingCustomer: Boolean?

    @get:com.fasterxml.jackson.annotation.JsonGetter("MaritalStatus")
    val maritalStatus: String?

    @get:com.fasterxml.jackson.annotation.JsonGetter("EmploymentStatus")
    val employmentStatus: String?

    override fun toContext(): com.gs.dmn.runtime.Context {
        val context = com.gs.dmn.runtime.Context()
        context.put("monthly", this.monthly)
        context.put("age", this.age)
        context.put("existingCustomer", this.existingCustomer)
        context.put("maritalStatus", this.maritalStatus)
        context.put("employmentStatus", this.employmentStatus)
        return context
    }

    fun equalTo(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as TApplicantData
        if (if (this.age != null) this.age != other.age else other.age != null) return false
        if (if (this.employmentStatus != null) this.employmentStatus != other.employmentStatus else other.employmentStatus != null) return false
        if (if (this.existingCustomer != null) this.existingCustomer != other.existingCustomer else other.existingCustomer != null) return false
        if (if (this.maritalStatus != null) this.maritalStatus != other.maritalStatus else other.maritalStatus != null) return false
        if (if (this.monthly != null) this.monthly != other.monthly else other.monthly != null) return false

        return true
    }

    fun hash(): Int {
        var result = 0
        result = 31 * result + (if (this.age != null) this.age.hashCode() else 0)
        result = 31 * result + (if (this.employmentStatus != null) this.employmentStatus.hashCode() else 0)
        result = 31 * result + (if (this.existingCustomer != null) this.existingCustomer.hashCode() else 0)
        result = 31 * result + (if (this.maritalStatus != null) this.maritalStatus.hashCode() else 0)
        result = 31 * result + (if (this.monthly != null) this.monthly.hashCode() else 0)
        return result
    }

    fun asString(): String {
        val result_ = StringBuilder("{")
        result_.append("Age=" + age)
        result_.append(", EmploymentStatus=" + employmentStatus)
        result_.append(", ExistingCustomer=" + existingCustomer)
        result_.append(", MaritalStatus=" + maritalStatus)
        result_.append(", Monthly=" + monthly)
        result_.append("}")
        return result_.toString()
    }

    companion object {
        @JvmStatic
        fun toTApplicantData(other: Any?): TApplicantData? {
            if (other == null) {
                return null
            } else if (other is TApplicantData?) {
                return other
            } else if (other is com.gs.dmn.runtime.Context) {
                var result_ = TApplicantDataImpl()
                result_.monthly = other.get("Monthly") as type.Monthly?
                result_.age = other.get("Age") as java.math.BigDecimal?
                result_.existingCustomer = other.get("ExistingCustomer") as Boolean?
                result_.maritalStatus = other.get("MaritalStatus") as String?
                result_.employmentStatus = other.get("EmploymentStatus") as String?
                return result_
            } else if (other is com.gs.dmn.runtime.DMNType) {
                return toTApplicantData(other.toContext())
            } else {
                throw com.gs.dmn.runtime.DMNRuntimeException(String.format("Cannot convert '%s' to '%s'", other.javaClass.getSimpleName(), TApplicantData::class.java.getSimpleName()))
            }
        }
    }
}
